raft/src/test/java/org/apache/kafka/raft/KafkaRaftClientTest.java

```diff
@@ -955,6 +977,55 @@ public void testInitializeAsCandidateAndBecomeLeaderQuorumOfThree(boolean withKi
             Arrays.asList(voterId, localId), record.key(), record.value());
     }
 
+    @ParameterizedTest
+    @ValueSource(booleans = { true, false })
+    public void testInitializeAsOnlyVoterWithEmptyElectionState(boolean withKip853Rpc) throws Exception {
+        int localId = randomReplicaId();
+        RaftClientTestContext context = new RaftClientTestContext.Builder(localId, Set.of(localId))
+            .withKip853Rpc(withKip853Rpc)
+            .build();
+        context.assertElectedLeader(1, localId);
+        assertEquals(0L, context.log.endOffset().offset());
+        assertTrue(context.client.quorum().isLeader());
+    }
+
+    @ParameterizedTest
+    @ValueSource(booleans = { true, false })
+    public void testInitializeAsFollowerAndOnlyVoter(boolean withKip853Rpc) throws Exception {
+        int localId = randomReplicaId();
+        RaftClientTestContext context = new RaftClientTestContext.Builder(localId, Set.of(localId))
+            .withKip853Rpc(withKip853Rpc)
+            .withElectedLeader(2, localId + 1)
+            .build();
+        context.assertElectedLeader(3, localId);
+        assertEquals(0L, context.log.endOffset().offset());
+        assertTrue(context.client.quorum().isLeader());
+    }
+
+    @ParameterizedTest
+    @ValueSource(booleans = { true, false })
+    public void testInitializeAsCandidateAndOnlyVoter(boolean withKip853Rpc) throws Exception {
+        int localId = randomReplicaId();
+        RaftClientTestContext context = new RaftClientTestContext.Builder(localId, Set.of(localId))
+            .withKip853Rpc(withKip853Rpc)
+            .withVotedCandidate(2, ReplicaKey.of(localId, ReplicaKey.NO_DIRECTORY_ID))
+            .build();
+        context.assertElectedLeader(2, localId);
+        assertTrue(context.client.quorum().isLeader());
+    }
+
+    @ParameterizedTest
+    @ValueSource(booleans = { true, false })
+    public void testInitializeAsResignedAndOnlyVoter(boolean withKip853Rpc) throws Exception {
+        int localId = randomReplicaId();
+        RaftClientTestContext context = new RaftClientTestContext.Builder(localId, Set.of(localId))
+            .withKip853Rpc(withKip853Rpc)
+            .withElectedLeader(2, localId)
+            .build();
+        context.assertElectedLeader(3, localId);
+        assertTrue(context.client.quorum().isLeader());
+    }
```

Comment:
Technically possible but it is odd that the replica is using a static voter set, with KIP_595_PROTOCOL and an elected leader or voted candidate that is not in the voter set.

Maybe it is less confusing if you limit these tests to KIP_853_PROTOCOL.

Minor but technically the protocol configuration is not needed since the replica doesn't need to send or handle RPCs to become leader.